benchmarks/ExecutionContextBenchmarks.cpp

```diff
@@ -0,0 +1,268 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "etl/ETLHelpers.hpp"
+#include "util/Random.hpp"
+#include "util/async/AnyExecutionContext.hpp"
+#include "util/async/AnyOperation.hpp"
+#include "util/async/context/BasicExecutionContext.hpp"
+#include "util/async/context/SyncExecutionContext.hpp"
+
+#include <benchmark/benchmark.h>
+
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <latch>
+#include <optional>
+#include <stdexcept>
+#include <thread>
+#include <vector>
+
+using namespace util;
+using namespace util::async;
+
+class TestThread {
+    std::vector<std::thread> threads_;
+    etl::ThreadSafeQueue<std::optional<uint64_t>> q_;
+    etl::ThreadSafeQueue<uint64_t> res_;
+
+public:
+    TestThread(std::vector<uint64_t> const& data) : q_(data.size() + 1), res_(data.size() + 1)
```

Comment:
I am trying to figure out why +1 here 🤔 .

Reply:
I think it was needed because `push_back` on the queue would hang since capacity is reached or something like that. I believe the code of the implementation changed since then and it's no longer needed 👍